store/disk.go

```diff
@@ -279,6 +325,38 @@ func (*OnDiskStoreBuilder) Delete(path, userID string) error {
 	return os.RemoveAll(storePath)
 }
 
+func (c *onDiskStore) readFromFallback(file *os.File) ([]byte, error) {
+	if pos, err := file.Seek(0, 0); err != nil {
+		return nil, err
+	} else if pos != 0 {
+		return nil, fmt.Errorf("failed to rewind file to start")
+	}
+
+	return c.fallback.Read(c.gcm, file)
+}
+
 func getEncryptedBlockSize(aead cipher.AEAD, blockSize int) int {
 	return blockSize + aead.Overhead()
 }
+
+func uint32ToByteArray(i uint32) []byte {
+	var b []byte
+	sh := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	sh.Len = 4
+	sh.Cap = 4
+	sh.Data = uintptr(unsafe.Pointer(&i))
+
+	return b[:]
+}
```

Comment:
probably better to use `encoding/binary` here (https://pkg.go.dev/encoding/binary)

Reply:
I found that, but that is variant based decoding.  I need a fixed value so I can always read the same amount of bytes in one go.

Reply:
Sorry. I misread the documentation. Fixed.